gst/edge/edge_sink.c

```diff
@@ -256,15 +256,18 @@ static void
 gst_edgesink_finalize (GObject * object)
 {
   GstEdgeSink *self = GST_EDGESINK (object);
-  if (self->host) {
-    g_free (self->host);
-    self->host = NULL;
-  }
 
-  if (self->edge_h) {
-    nns_edge_release_handle (self->edge_h);
-    self->edge_h = NULL;
-  }
+  g_free (self->host);
+  self->host = NULL;
+
+  g_free (self->dest_host);
+  self->dest_host = NULL;
+
+  g_free (self->topic);
+  self->topic = NULL;
+
+  nns_edge_release_handle (self->edge_h);
+  self->edge_h = NULL;
```

Comment:
```suggestion
  if (self->edge_h) {
    nns_edge_release_handle (self->edge_h);
    self->edge_h = NULL;
  }
```
```nns_edge_release_handle``` definitely requires a null check. Otherwise, it complains about that using ```g_critical``` or the equivalent.